packages/flutter_tools/lib/src/vmservice.dart

```diff
@@ -1249,18 +1249,73 @@ class ServiceMap extends ServiceObject implements Map<String, dynamic> {
   @override
   void addAll(Map<String, dynamic> other) => _map.addAll(other);
   @override
+  // TODO(srawlins): Dart 2.0 requires this method to be implemented.
+  // ignore: override_on_non_overriding_method
+  void addEntries(Iterable<Object> entries) {
+    // Change Iterable<Object> to Iterable<MapEntry<K, V>> when
+    // the MapEntry class has been added.
+    throw new UnimplementedError('addEntries');
+  }
+  @override
+  // TODO(srawlins): Dart 2.0 requires this method to be implemented.
+  // ignore: override_on_non_overriding_method
+  Map<K2, V2> cast<K2, V2>() {
+    throw new UnimplementedError('cast');
+  }
+  @override
   void clear() => _map.clear();
   @override
   bool containsValue(dynamic v) => _map.containsValue(v);
   @override
   bool containsKey(Object k) => _map.containsKey(k);
   @override
+  // TODO(srawlins): Dart 2.0 requires this method to be implemented.
+  // ignore: override_on_non_overriding_getter
+  Iterable<Null> get entries {
+    // Change Iterable<Null> to Iterable<MapEntry<K, V>> when
+    // the MapEntry class has been added.
+    throw new UnimplementedError('entries');
+  }
+  @override
   void forEach(void f(String key, dynamic value)) => _map.forEach(f);
   @override
+  // TODO(srawlins): Dart 2.0 requires this method to be implemented.
+  // ignore: override_on_non_overriding_method
+  Map<K2, V2> map<K2, V2>(Object transform(String key, dynamic value)) {
+    // Change Object to MapEntry<K2, V2> when
+    // the MapEntry class has been added.
+    throw new UnimplementedError('map');
+  }
+  @override
   dynamic putIfAbsent(String key, dynamic ifAbsent()) => _map.putIfAbsent(key, ifAbsent);
   @override
   void remove(Object key) => _map.remove(key);
   @override
+  // TODO(srawlins): Dart 2.0 requires this method to be implemented.
+  // ignore: override_on_non_overriding_method
```

Comment:
shouldn't this patch actually implement these?

Reply:
It doesn't have to. For example, if its going to delegate down to other `Map` objects, which don't have implementations of these methods, then it can't fully implement.

However, 28 days later, those methods have been implemented on Map, so someone could write an implementation.

This PR was intended to avoid errors around ServiceMap claiming to implement Map, and yet missing several methods. I guess y'all aren't encountering such errors?

Reply:
I'm not aware of any...